docs/madr/decisions/035-mesh-metric-policy.md

```diff
@@ -0,0 +1,126 @@
+policies to point to specific DPs. I imagine there would be a default wide policy (possibly `Mesh` scoped) and per app / framework / language overrides.
+2. What I also don’t like about this is that “filters” are under envoy - I’d probably rename it to filter and introduce 
+the concept of “profiles” like we discussed. Maybe something like this:
+
+```yaml
+filter:
+  usedOnly: true # true or false
+  profile: minimal # one of minimal, default, full
+  regex: http2_act.* # only profile or regex can be defined
+```
+3. Should we also think about grouping the settings somehow, tls/port/path/tags/envoy are all “mesh/sidecar” metrics and 
+“aggregate” are application metrics, or maybe we should just rename `aggregate` to `otherContainers`?
+
+#### What other meshes do
+- Istio has [Telemetry](https://istio.io/latest/docs/reference/config/telemetry/#Metrics) policy that allows defining providers (our backends), overrides (that allows disabling / changing tags), reporting interval
+- LinkerD has only tutorials on how to [export metrics](https://linkerd.io/2.14/tasks/exporting-metrics/#proxy), and creating a service profile can enable per route metrics
+
+#### New API suggestion:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshMetrics
+metadata:
+  name: default
+  namespace: kuma-system
+spec:
+  targetRef:
+    kind: Mesh | MeshSubset | MeshService | MeshServiceSubset
+  default:
+    filter:
+      profile: minimal
+      regex: http2_act.* # only profile or regex can be defined
+      usedOnly: true
+    aggregate:
+      - path: "/metrics/prometheus" # is name still needed since targetRef can just point to a name?
+        port: 8888
+      - port: 8000 # default path is /metrics
+    backends:
+      - type: Prometheus
+        prometheus:
+          port: 5670
+          path: /metrics
+          tls:
+            mode: disabled | providedTLS | activeMTLSBackend
+      - type: OpenTelemetry
+        openTelemetry:
+          endpoint: otel-collector:4317
+```
+
+### Configure MADS based on policy data
+MADS server is responsible for aggregating information about dataplanes needed for Prometheus monitoring. 
+It now extracts information from the `Mesh` resource which proxy should be monitored and creates a `MonitoringAssignment` response.
+
+We need to change the source of our information about monitored proxies from `Mesh` to `MeshMetric` policy. We cannot treat 
+new policy as usual in this context, we should not merge policies, but create different `MonitoringAssignment` for each 
+defined policy, this is needed for scenarios like migrating to a new backend. For example, when we define two `MeshMetric` 
+policies that target the same data plane proxy, we should create two separate `MonitoringAssignment` responses. 
+We can achieve this by extending Prometheus integration, so that it sends backend name with requests to MADS API.
```

Comment:
Ideally we shouldn't modify MADS. If we need to maybe we should consider using https://prometheus.io/docs/prometheus/latest/configuration/configuration/#http_sd_config instead

Feels like we should have an identifier of some sort for the request for a backend so that it matches the correct backend.

Reply:
Let's discuss with @jakubdyszkiewicz but I think mads v1 was pretty painful to contribute.

Reply:
I'm trying to wrap my head around this. Is the strategy then to _name_ the backend and put it in Prometheus configuration? Schema above does not have a name.

Reply:
About your last comment, it was in my head, but I forgot about it, added now